sdk/include/opentelemetry/sdk/metrics/exemplar/reservoir_cell.h

```diff
@@ -0,0 +1,139 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+#ifndef ENABLE_METRICS_PREVIEW
+#  include <cstddef>
+#  include <memory>
+#  include <vector>
+#  include "opentelemetry/common/timestamp.h"
+#  include "opentelemetry/context/context.h"
+#  include "opentelemetry/nostd/shared_ptr.h"
+#  include "opentelemetry/sdk/common/attribute_utils.h"
+#  include "opentelemetry/sdk/metrics/data/exemplar_data.h"
+#  include "opentelemetry/sdk/metrics/exemplar/filter.h"
+#  include "opentelemetry/trace/context.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace metrics
+{
+/**
+ * A Reservoir cell pre-allocated memories for Exemplar data.
+ */
+class ReservoirCell
+{
+public:
+  ReservoirCell() = default;
+
+  /**
+   * Record the long measurement to the cell.
+   */
+  void RecordLongMeasurement(long value,
+                             const MetricAttributes &attributes,
+                             const opentelemetry::context::Context &context)
+  {
+    long_value_ = value;
+    offerMeasurement(attributes, context);
+  }
+
+  /**
+   * Record the long measurement to the cell.
+   */
+  void RecordDoubleMeasurement(double value,
+                               const MetricAttributes &attributes,
+                               const opentelemetry::context::Context &context)
+  {
+    double_value_ = value;
+    offerMeasurement(attributes, context);
+  }
+
+  /**
+   * Retrieve the cell's {@link ExemplarData}.
+   *
+   * <p>Must be used in tandem with {@link #recordLongMeasurement(long, Attributes, Context)}.
+   */
+  std::shared_ptr<ExemplarData> GetAndResetLong(const MetricAttributes &point_attributes)
```

Comment:
nit - can it return unique_ptr to the ExemplarData, as I don't think this data would be used anywhere else after reset ?

Reply:
thanks using unique_ptr now.